[Luoq1-Xu] iP

Luoq1-Xu opened the pull request:
## Nyx 🍰 

> When ChatGPT gets so lost you need to start a new chat

Looking for the next **DeepSeek R1**? Sorry, you won't find it here. However, what you will find is a nice little task manager chatbot. It won't forget your tasks, no matter how many messages you send :)

- Intuitive interface
- Easy to use
- Basic task management functionality

To get started:

1. Go to the [releases page](https://github.com/Luoq1-Xu/ip/releases)
2. Find the latest release, and download `Nyx.jar`
3. Go to your terminal, cd into the directory which contains `Nyx.jar` and enter 
```java
java -jar Nyx.java
```
4. Enjoy! 😋 

Features
- [ ] Three types of tasks - todo, deadline and event
- [ ] Persistent memory (your tasks are automatically saved and persist even after you close the app!)
- [X] Ability to mark and unmark tasks

```
Are we there yet?
```

`Initializing...`

```java
public static void main(String[] args) {
        new Nyx().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Luoq1-Xu committed:
rename chatbot

Luoq1-Xu committed:
Add echo functionality
Rename to Nyx

Luoq1-Xu committed:
Add task.java class
Add list functionality to Nyx.java

Luoq1-Xu committed:
Add ability to mark tasks as complete and incomplete

Luoq1-Xu committed:
add TodoTask.java
add DeadlineTask.java
add EventTask.java
add three types of task tracking functionality to Nyx.java

Luoq1-Xu committed:
Add automated text ui testing functionality

Luoq1-Xu committed:
Add NyxException class
Add InvalidUsageException class
Add UnknownCommandException class
Add basic error handling to Nyx.java

Luoq1-Xu committed:
Add delete task functionality

Luoq1-Xu committed:
Add enum for CommandType

Luoq1-Xu committed:
Add tasks.txt
Add save functionality to Nyx.java to save and load from file

Luoq1-Xu committed:
Merge branch 'branch-Level-7'

Luoq1-Xu committed:
Add Date handling functionality to nyx.java

Luoq1-Xu committed:
Merge branch 'branch-Level-8'
Add Date handling functionality to Nyx.java

Luoq1-Xu committed:
Refactor commands into classes
Add Ui, Storage, Parser and TaskList classes
Update automated text testing

Luoq1-Xu committed:
Merge branch 'branch-A-MoreOOP'
Refactor code to use more OOP

Luoq1-Xu committed:
organize class into commands, exceptions and tasks packages

Luoq1-Xu committed:
Merge branch 'branch-A-Packages'
Organize classes into packages: commands, tasks, exceptions

Luoq1-Xu committed:
Merge branch 'add-gradle-support'
Add gradle support

Luoq1-Xu committed:
Add unit tests for Parser and DeadlineCommand

Luoq1-Xu committed:
Merge branch 'branch-A-JUnit'
Add unit tests for Parser and DeadlineCommand

Luoq1-Xu committed:
Create JAR file

Luoq1-Xu committed:
Merge branch 'branch-A-Jar'
Create JAR file

Luoq1-Xu committed:
Add javaDocs to Nyx.java
Add javaDocs to Parser.java
Add javaDocs to Storage.java
Add javaDocs to TaskList.java
Add javaDocs to Ui.java

Luoq1-Xu committed:
Add javaDocs for Tasks package

Luoq1-Xu committed:
Add javaDocs for Exceptions package

Luoq1-Xu committed:
Add javaDocs to Commands package

Luoq1-Xu committed:
Add slight changes to follow Coding Standard

Files changed:
* Nyx.java
* Storage.java
* EventCommand.java
* TodoCommand.java
* DeadlineCommandTest.java

Luoq1-Xu committed:
Add find functionality

Add find function that allows user to search for tasks
with a matching keyword

Luoq1-Xu committed:
Merge branch 'branch-A-JavaDoc'
Add javaDocs for all packages in nyx

Luoq1-Xu committed:
Merge branch 'branch-A-CodingStandard'

Luoq1-Xu committed:
Merge branch 'branch-Level-9'

Add find functionality

# Conflicts:
#	src/main/java/nyx/TaskList.java

Luoq1-Xu committed:
Make changes to align to checkstyle

Luoq1-Xu committed:
Merge branch 'branch-A-CheckStyle'
Make slight changes to align to checkstyle

Luoq1-Xu committed:
Add GUI support

Luoq1-Xu committed:
Merge branch 'branch-Level-10'

Add GUI support

Luoq1-Xu committed:
Change DialogBox and mainWindow font to lucida sans
Fix some checkstyle issues

DanielJames0302 commented on src/main/java/nyx/Parser.java:
```diff
@@ -0,0 +1,66 @@
+            return new FindCommand(input);
+        } else {
+            return new UnknownCommand();
+        }
```
Perhaps change to switch case for better code readability ?

Luoq1-Xu replied:
Will consider it, thanks!

DanielJames0302 commented on src/main/java/nyx/Storage.java:
```diff
@@ -0,0 +1,136 @@
+     * If the file or directories do not exist, they are created.
+     *
+     * @return An ArrayList of tasks loaded from the file.
+     */
```
Good javadoc comments for this place and other places

DanielJames0302 commented on src/main/java/nyx/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,65 @@
+     *
+     * @return The created DeadlineTask.
+     */
+    private DeadlineTask getDeadlineTask() {
```
Good method naming

DanielJames0302 commented on src/main/java/nyx/tasks/Task.java:
```diff
@@ -0,0 +1,85 @@
+     *
+     * @return 1 if the task is completed, 0 otherwise.
+     */
+    public int getCompletedStatus() {
```
Perhaps more boolean name for this method ?

Luoq1-Xu replied:
Yep, will think of a better name

DanielJames0302 reviewed (commented):
LGTM. Just some nits to fix.

Luoq1-Xu committed:
Add assertions in Command and Nyx classes

The Command and Nyx classes lack state validations through assertions.
This may propagate into an invalid state, impacting reliability.
Let's add assertions to the Command and Nyx classes. This change
enforces critical conditions during execution, improving early error
detection.
These targeted assertions help to catch unexpected behaviour early
during development.

Luoq1-Xu committed:
Add spacing to improve logical structure
Add more comments for readability
Refactor Parser to use switch

Parser.java exhibits a long chain of if/else statements that reduce
readability and clarity. Some files also lack sufficient comments,
which obscures the intent behind some design decisions.

A refactoring is essential to improve maintainability and ease of
comprehension. Let's add spacing to visually separate code sections,
replace if/else chains with a switch statement to streamline logic,
and add comments to clarify the purpose of critical code segments.

Using a switch statement produces a more organized structure, while
readability improvements and comments help future developers understand
the code with minimal effort.

Luoq1-Xu committed:
Merge pull request #1 from Luoq1-Xu/branch-A-Assertions

Add assertions in Command and Nyx classes

Luoq1-Xu committed:
Merge branch 'master' of https://github.com/Luoq1-Xu/ip

Luoq1-Xu committed:
Merge branch 'master' into branch-A-CodeQuality

Luoq1-Xu committed:
Merge pull request #3 from Luoq1-Xu/branch-A-CodeQuality

Add spacing to improve logical structure

Luoq1-Xu committed:
Add ability to tag tasks
Refactor storage to handle new save format

Luoq1-Xu committed:
Merge branch 'branch-C-Tagging'

Luoq1-Xu committed:
Add retro styling to GUI

Luoq1-Xu committed:
Improve GUI

add help command

Luoq1-Xu committed:
Merge branch 'branch-A-BetterGui'

Luoq1-Xu committed:
Add more personality with title and welcome message

Luoq1-Xu committed:
Merge branch 'branch-A-Personality'

Luoq1-Xu committed:
Add user guide

Luoq1-Xu committed:
Merge branch 'branch-A-UserGuide'

Luoq1-Xu committed:
Update README.md

Fixed incorrect example for creating an event task

Luoq1-Xu committed:
Add acknowledgments for image use and credit for AI tool usage

Luoq1-Xu committed:
Merge branch 'branch-credit'

Luoq1-Xu committed:
Fix incorrect example in documentation
